registry/storage/blobcachemetrics.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"expvar"
+	"sync/atomic"
+
+	"github.com/docker/distribution/registry/storage/cache"
+)
+
+type blobStatCollector struct {
+	metrics cache.Metrics
+}
+
+func (bsc *blobStatCollector) Hit() {
```

Comment:
You can also just use the prometheus client library. Granted, it's prometheus specific but it does exactly what you want and it exposes the metrics both as protobuf (for which the schema as well as generated code is available here: https://github.com/prometheus/client_model) and as a simple to parse text based format.

I'm still working on the expvar exporter and it gets pretty nasty to find a good mapping/config format to translate the non-structured expvars. However we expose metrics, it would be good to have some stable structure, for example not mixing metrics with config settings and having the same level in the json mean the same things. So not mixing maps and lists at the same layer like registry.cache.storage vs registry.notifications.endpoints etc.